File: windows/PhoneWheel.Server/Models/ButtonPacket.cs
namespace PhoneWheel.Server.Models;

/// <summary>
/// Representa o pacote de evento de botão definido no protocolo PhoneWheel
/// (ver <c>protocol/protocol.md</c>), enviado pelo aplicativo Android via UDP.
///
/// Campos:
/// - <see cref="Type"/>: identificador fixo do pacote ("button").
/// - <see cref="Button"/>: identificador lógico do botão (0 = A, 1 = B, ...).
/// - <see cref="Pressed"/>: true para pressionado, false para liberado.
/// - <see cref="Timestamp"/>: instante de criação do pacote no dispositivo Android,
///   em milissegundos desde a época Unix.
///
/// Este modelo é apenas uma estrutura de dados e não possui lógica de rede,
/// desserialização ou processamento.
/// </summary>
public sealed record ButtonPacket
{
    public const string TypeButton = "button";

    public string Type { get; init; } = TypeButton;

    public int Button { get; init; }

    public bool Pressed { get; init; }

    public long Timestamp { get; init; }
}

File: android/app/src/main/java/com/phonewheel/model/ButtonPacket.kt
package com.phonewheel.model

/**
 * Representa o pacote de evento de botão definido no protocolo PhoneWheel
 * (ver `protocol/protocol.md`).
 *
 * Campos:
 * - [type]: identificador fixo do pacote ("button").
 * - [button]: identificador lógico do botão (0-9 no mapeamento Standard Gamepad).
 * - [pressed]: true para pressionado, false para liberado.
 * - [timestamp]: instante de criação do pacote, em milissegundos desde a época Unix.
 *
 * Este modelo é apenas uma estrutura de dados (data class) e não possui
 * lógica de rede, serialização ou acesso a sensores.
 */
data class ButtonPacket(
    val button: Int,
    val pressed: Boolean,
    val timestamp: Long = System.currentTimeMillis(),
    val type: String = TYPE_BUTTON
) {
    companion object {
        const val TYPE_BUTTON = "button"
    }
}